timely_beliefs/beliefs/probabilistic_utils.py

```diff
@@ -539,8 +539,41 @@ def get_belief_at_cumulative_probability(
         return df2.head(1)
 
 
-def get_mean_belief(df: "classes.BeliefsDataFrame") -> "classes.BeliefsDataFrame":
-    """Convenience function to select the expected value."""
+def get_mean_belief(
+    df: "classes.BeliefsDataFrame", distribution: str = "uniform"
+) -> "classes.BeliefsDataFrame":
+    """Convenience function to select the expected value.
+    Assumes the data frame contains a single belief per event.
```

Comment:
Not sure if this is correct. As you are using it, `df`contains one (probabilistic) belief, and maybe that is what we should be saying here (and also assert if I'm correct).

In line 567, it seems you are using the length of `df` to decide the cumulative_probability. I might misinterpret what's going on, but it seems to me that makes sense only if `df` contains only one belief, and not many?

Reply:
One (probabilistic) belief per event is correct, I think.

- In line 551, I start to loop over the events (that is, loop over the beliefs, because there is one belief per event) to extract one list of values and one list of cumulative probabilities, for each event (/ for each belief).
- From these, in line 556, I create a distribution function for each event (/ for each belief).
- In line 563, I reduce the length of the data frame, from n_events * average_number_of_probablistic_values_per_event, to n_events * 1
- In line 564, I overwrite the cp index level (of the now smaller data frame) to cp=0.5. Technically, the mean value could be located at a different cp value, so this does need a fix. Something along the lines of:
  `dist.computeCDF(dist.getMean())`
  I'll get on it.
- Finally, in line 569, I overwrite the event_values column with the mean value for each event (/ for each belief).